### PR TITLE
LIRC driver: support for save remote controls via config msg.

### PR DESCRIPTION
**Pull request by hpsaturn**
**Prerequisite**:
For testing this PR, please add in `/etc/lirc/lircd.conf` or create `lircd.conf` with:
```
include "/etc/lirc/lircd.matrix.conf"
```
**Demo testing**:
Run nodejs demo (/src/js_test):
```
nodejs test_ir_remote.js
```

### CODE REVIEW

File: src/driver_lirc.cpp [hunk 1]
```diff
@@ -30,11 +31,41 @@ const bool kLircDriverDebugEnabled = false;
 bool LircDriver::ProcessConfig(const DriverConfig& config) {
   LircParams lirc(config.lirc());
 
+  if (lirc.config() != "") {
```
Comment (arhuaco) [inline src/driver_lirc.cpp]: There's an "empty()" method for strings. I think it should be used here. If ```!lirc.config().empty()```.

File: src/driver_lirc.cpp [hunk 2]
```diff
@@ -30,11 +31,41 @@ const bool kLircDriverDebugEnabled = false;
...
+    // LIRC remotes config from MOS via proto
+    std::ofstream remotes_config("/etc/lirc/lircd.matrix.conf");
+    remotes_config << lirc.config();
+    remotes_config.close();
+
+    if (kLircDriverDebugEnabled) {
+      std::cout << "new remote database saved" << std::endl;
```
Comment (arhuaco) [inline src/driver_lirc.cpp]: Debug messages should go to stderr!

File: src/driver_lirc.cpp [hunk 3]
```diff
@@ -30,11 +31,41 @@ const bool kLircDriverDebugEnabled = false;
...
+      std::cout << "LIRC service restart done." << std::endl;
+    }
+    return true;
+  }
+
+  // execute commands over remote device
   if (lirc.device() == "" || lirc.command() == "" ||
       !isValidLircSymbol(lirc.device()) || !isValidLircSymbol(lirc.command())) {
-    zmq_push_error_->Send(
+    std::string msg_error =
```
Comment (arhuaco) [inline src/driver_lirc.cpp]: const

File: src/driver_lirc.cpp [hunk 4]
```diff
@@ -30,11 +31,41 @@ const bool kLircDriverDebugEnabled = false;
...
         std::string(kLircDriverName) +
-        " error: Device or command parameter is missing or invalid.");
+        " error: Device or command parameter is missing or invalid.";
+    std::cout << msg_error << std::endl;
```
Comment (arhuaco) [inline src/driver_lirc.cpp]: cerr

File: src/js_test/test_ir_remote.js [hunk 5]
```diff
@@ -20,17 +20,68 @@ var zmq = require('zmq')
...
+var http = require('http');
+var fs = require('fs');
+
+/**
+ * LIRC remote database on admobilize assets.
+ * Orginal source on: https://sourceforge.net/p/lirc-remotes/code/ci/master/tree/
+ */
+
+// Example remote device (SONY control RM-AAU014)
+var remote_url="http://assets.admobilize.com/lirc-remotes/sony/RM-AAU014.lircd.conf"
```
Comment (arhuaco) [inline src/js_test/test_ir_remote.js:31]: Is it OK to download from here all the time? Maybe we should copy the file and download from one of our servers. Just wondering.

We can leave this as it is for now.

File: src/driver_lirc.cpp [hunk 6]
```diff
@@ -2,6 +2,7 @@
  * Copyright 2016 <Admobilize>
  * MATRIX Labs  [http://creator.matrix.one]
  * This file is part of MATRIX Creator MALOS
+ * Author: @hpsaturn
```
Comment (arhuaco) [inline src/driver_lirc.cpp:5]: Are we using the Author thing? I think we leave ADM and that's it. Not sure. You might want to ask Brian.

Comment (hpsaturn) [reply src/driver_lirc.cpp:5]: Always the author has the patrimonial rights

File: docs/lirc.md [hunk 7]
```diff
@@ -47,13 +47,15 @@ This driver reports errors when an invalid configuration is sent.
...
+This driver need at last one LIRC remote configuration for send IR commands to any device. MALOS LIRC driver support multiple remote controls in the same file and you can use a previous remote control recorded. You will can download this remotes config from MATRIX assets repository and then send it via `config` field of `LircParams` protobuf. For more details plese see JavaScript example described below.
 
+
+**Example for SONY RM-AAU014**:
 [http://assets.admobilize.com/lirc-remotes/sony/RM-AAU014.lircd.conf]("http://assets.admobilize.com/lirc-remotes/sony/RM-AAU014.lircd.conf")
 
 ### Read
 
-This driver for now not return any data via read MALOS protocol. 
+This driver for now not return any data via read MALOS protocol only execute commands and set configurations.
```
Comment (arhuaco) [inline docs/lirc.md]: This driver doesn't any data using MALOS protocol. It only executes commands and sets configurations.

File: docs/lirc.md [hunk 8]
```diff
@@ -1 +1,184 @@
...
+This driver NOT needs keep-alive messages [as specified in the MALOS protocol](https:////github.com/matrix-io/matrix-creator-malos/blob/master/README.md#keep-alive-port).
+
+
+### Errors
+
+This driver reports errors when an invalid configuration is sent.
+
+
+### Write
+
```
Comment (arhuaco) [inline docs/lirc.md:46]: You should be consistent with the amount of newlines after each title.

**Review by arhuaco — APPROVED** [review]

**Pull request merged**